app/src/main/java/com/github/alfabravo2013/readyforexams/presentation/landing/LandingFragment.kt

```diff
@@ -0,0 +1,58 @@
+package com.github.alfabravo2013.readyforexams.presentation.landing
+
+import android.os.Bundle
+import androidx.fragment.app.Fragment
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.fragment.app.viewModels
+import androidx.navigation.fragment.findNavController
+import androidx.viewpager2.adapter.FragmentStateAdapter
+import com.github.alfabravo2013.readyforexams.R
+import com.github.alfabravo2013.readyforexams.databinding.FragmentLandingBinding
+import com.google.android.material.tabs.TabLayoutMediator
+
+class LandingFragment : Fragment() {
+    private val viewModel: LandingViewModel by viewModels()
+
+    private var _binding: FragmentLandingBinding? = null
+    private val binding: FragmentLandingBinding get() = _binding!!
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        _binding = FragmentLandingBinding.inflate(inflater, container, false)
+        return binding.root
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        val adapter = ViewPagerAdapter(this)
+        binding.landingViewPager.adapter = adapter
+        TabLayoutMediator(binding.landingTabLayout, binding.landingViewPager) { _, _ -> }.attach()
+
+        binding.landingLoginButton.setOnClickListener {
+            findNavController().navigate(R.id.action_landingFragment_to_loginFragment)
+        }
+    }
+
+    override fun onDestroyView() {
+        super.onDestroyView()
+        _binding = null
+    }
+}
+
+class ViewPagerAdapter(fragment: Fragment) : FragmentStateAdapter(fragment) {
+    companion object {
+        private const val TOTAL_PAGES = 3
+    }
+
+    override fun getItemCount(): Int = TOTAL_PAGES
+
+    override fun createFragment(position: Int): Fragment {
+        return when (position) {
+            in 0..2 -> FirstOnboardingFragment()
```

Comment:
you do not need this. You can just return FirstOnboardingFragment. Because total PAGE is 3 it will be rendered 3 times.